components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -141,6 +141,55 @@ def fileread_gen(fin, fsize, bufsize):
     fin.close()
 
 
+def countAnnotations(annotationlinks=None):
+    """
+    Count the different (unique) annotions from the
+    provided annotation links
+    """
+
+    counts = {
+        "TagAnnotation": 0,
+        "FileAnnotation": 0,
+        "CommentAnnotation": 0,
+        "LongAnnotation": 0,
+        "MapAnnotation": 0,
+        "OtherAnnotation": 0}
+
+    if annotationlinks is None:
+        return counts
+
+    atypes = {
+        omero.model.TagAnnotationI: "TagAnnotation",
+        omero.model.FileAnnotationI: "FileAnnotation",
+        omero.model.CommentAnnotationI: "CommentAnnotation",
+        omero.model.LongAnnotationI: "LongAnnotation",
+        omero.model.MapAnnotationI: "MapAnnotation"}
+
+    uniqueIds = []
+    total = 0
+    regAnnotations = 0
+    for al in annotationlinks:
```

Comment:
as Annotation ids are in the same table I would assume they shouldn't ever have the same id regardless of type, so unique list could be constructed from the annotationlinks. I think there is no need to check.

Then you could just do: `uniqueIds = set ([al._child._id._val for al in annotationlinks])`